RouteComposer/Classes/Actions/UISplitViewController+Action.swift

```diff
@@ -83,4 +87,39 @@ public struct SplitViewControllerActions {
         }
     }
 
+    /// Pushes a view controller *onto* the detail stack in the `UISplitViewController`, where the detail is a `UINavigationController`
+    public struct PushOnToDetailsAction<ViewController: UISplitViewController>: ContainerAction {
+
+      /// Constructor
+      init() {
+      }
+
+      public func perform(embedding viewController: UIViewController, in childViewControllers: inout [UIViewController]) throws {
+        guard !childViewControllers.isEmpty else {
+          throw RoutingError.compositionFailed(.init("Master view controller is not set in " +
+            "UISplitViewController to push on a detail view controller \(viewController)."))
+        }
+        childViewControllers.append(viewController)
```

Comment:
@benronicus I have a doubt here tbh. 
This action requires a `UINavigationController` to be the last in stack to be successfully performed. See `perform(with ,on, animated, completion:)`,
But this method that is used for the instant build of the whole `UISplitViewController` doesn't have this check. So it will create a situation when using the same action may behave differently. I think the `viewControllers as? UINavigationController` should be added along with the check that there are not more than 2 `UINavigationController`s in the `childViewControllers` stack. But i am not hugely familiar with the split controller and its sideeffects. So i am curious about your thoughts about it. I would prefer action to not allow to shoot in to the leg, basically.

Reply:
Apologies for the slow response, just seeing this. I've been using the action without issue but  I'll give it some more thought and come back with a new PR if I see improvements to be made. Can you clariy what you mean by "this method that is used for the instant build"?

Reply:
@benronicus sorry. English is not my native language. So it means that this method will be used by the router to build the whole view controller stack from scratch. Imagine the situation that you need to build navigation controller with 2 view controllers inside. Router will call this method twice saying i have this view controller I don’t know how to build it into the array of view controllers that i will later put into your navigation controller. 

So basically the nature of your actions than it expects split view controller to be build out of the navigation controllers. And it should not let other view controllers in the stack and throw the exception. Similar how the method below does it

Reply:
@benconstantine39 @benronicus I actually added this change myself after i merged your PR. So you can see what i meant in the latest version